sdf/dataframe.mdx

```diff
@@ -0,0 +1,135 @@
+---
+title: Arrow Dataframe state
+description: Stateful Dataflows Dataframe Integration
+sidebar_position: 100
+---
+
+States are one of the core concepts in stateful dataflows.  States can be partitioned by key which is used to divided state into multiple partitions.  
+
+Each partition state can have many different types.  One of the type is `arrow-row`.  For example, follow snippe define state `count_per_word` that store each row into arrow dataframe.
+
+```yaml
+    states:
+      count-per-word:
+        type: keyed-state
+        properties:
+          key:
+            type: string
+          value:
+            type: arrow-row
+            properties:
+              count: 
+                type: u32
+```
+
+
+In here, we are defining a state `count_per_word` to track frequency of the words.  For each key, we have a row that has a column `count` that store the count of the word.
+For example, let's we have following word: `apple`, `orange`, `banana`, `orange`, `grape`, `orange`, `banana`.  
+
+Then this will be mapped to arrow dataframe as follows:
+
+| _key | count |
+| --- | --- |
+| apple | 2 |
+| orange | 3 |
+| banana | 2 |
+| grape | 1 |
+
+
+To update the state, you can use the `update-state` operator as below:
+
+```yaml 
+update-state:
+    run: |
+    fn count_word(_word: String) -> Result<()> {
+        let mut state = count_per_word();
+        state.count += 1;
+        state.update()?;
+        Ok(())
+    }
+```
+
+Note that state value can be access using `count_per_word` state function which is automatically injected by SDF builder.
+
+Most operator like `update-state`, operator, state function only return value of the partition state. 
```

Comment:
This API is invoked by the `update-state` operator, which only returns the value of the partition state.